### PR TITLE
[DEV-11487] Remove deprecated methods 

### PR DESCRIPTION
**Pull request by nateshim-indico**
## Summary
Removing methods marked as deprecated in previous releases

## Checklist:

- [x] Jira story/task is put into PR title
- [x] Code has been self-reviewed
- [x] Hard-to-understand areas have been commented
- [x] Documentation has been updated
- [x] My changes generate no new warnings
- [x] Unit tests have been added for base functionality and known edge cases
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/integration/queries/test_workflow.py [hunk 1]
```diff
@@ -515,12 +543,22 @@ def _new_dataset_for_updating(client):
...
     )
 
+    questionnaire = client.call(
+        GetQuestionnaire(
+            questionnaire.model_group_by_name(
+                classifier_name
+            ).model_group.questionnaire_id
+        )
+    )
+    while questionnaire.questions_status == "STARTED":
```
Comment (nateshim-indico) [inline tests/integration/queries/test_workflow.py:553]: I think GetQuestionnaire should be a `RequestChain` with `wait` kwarg - previously CreateQuestionnaire handled the waiting but AddModelComponent does not wait on Questionnaire status which could cause confusion because a Questionnaire num_examples could be off when calling GetQuestionnaire right after adding model component.

**Comment by goatrocks** [comment]
You'll want to write up some release notes that call out each function and their alternative as part of this work if that isn't done already :)

**Review by jacobmanderson — APPROVED** [review]
so much red

**Review by pastap — APPROVED** [review]
beautiful

File: tests/integration/queries/test_workflow.py (hunk 1 above)
Comment (nateshim-indico) [reply tests/integration/queries/test_workflow.py:553]: will make separate pr for this
